packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1065,9 +1065,10 @@ export class AccountHandler {
     await this.customs.check(request, email, 'accountStatusCheck');
 
     try {
-      const exist = await this.db.accountExists(email);
+      const account = await this.db.accountRecord(email);
       return {
-        exists: exist,
+        exists: !!account,
+        verified: !!account && account.verifierSetAt > 0,
```

Comment:
@vbudhram do you have an idea of how unsafe this additive change is?  If existing clients/RPs use the response without doing any sort of validation on its exact shape, it should be all right I'd think.  Also, this would expose the verified status of an email, which I'm not quite sure about.

Reply:
> do you have an idea of how unsafe this additive change is?

I'm not actually sure how many RP/clients use this endpoint. It is currently used on the [FxA frontend](https://github.com/mozilla/fxa/blob/9fd866c8cf02c658d4cebd4e34a4c4dd9589e817/packages/fxa-content-server/app/scripts/views/index.js#L237) to help redirect users to signin/signup if email exists, but that is as far as I can tell.

In FxA, we have had a tendency to overload the meaning of `verified` (session, email, account) and it always get confusing. Using `verifierSet` as response might make things a little more clear.

> Also, this would expose the verified status of an email

We currently have `/account/profile` and `/recovery_email/status` endpoints that exposes similar data, but they require a session token. This endpoint is unauthenicated but does have rate limiting in place.

I'm inclined to say that this is probably ok, if an attacker knew that no password was set, best they could do is force a password reset.